Fix remove unnecessary overview headings (628) (#632)
astronomer/docs: astro/kubernetespodoperator.md

id: kubernetespodoperator
---

> The [KubernetesPodOperator](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html) is one of the most powerful Apache Airflow operators. Similar to the Kubernetes Executor, this operator works with the Kubernetes API to dynamically launch a Pod in Kubernetes for each task that needs to run and terminates each Pod once the task is complete. This results in an isolated, containerized execution environment for each task that is separate from tasks otherwise being executed by Celery workers. 

Reviewer: ```suggestion
The [KubernetesPodOperator](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html) is one of the most powerful Apache Airflow operators. Similar to the Kubernetes executor, this operator dynamically launches a Pod in Kubernetes for each task and terminates each Pod once the task is complete. This results in an isolated, containerized execution environment for each task that is separate from tasks otherwise being executed by Celery workers.
```

Tried to cut out some implementation details here that in retrospect I think are not needed

> ## KubernetesPodOperator benefits

Reviewer: ```suggestion
## Benefits of the KubernetesPodOperator
```

Let's keep this consistent -- the deferrable operator doc says `Benefits of deferrable operators`.

The KubernetesPodOperator enables you to:

> - Execute a custom Docker image per task with Python packages and dependencies that would otherwise conflict with the rest of your Deployment's dependencies. This includes Docker images in a private registry or repository.

Reviewer: ```suggestion
- Execute a custom Docker image per task with Python packages and dependencies that would otherwise conflict with the rest of your Deployment's dependencies. This includes Docker images in a private registry or repository.
- Run tasks Kubernetes cluster outside of the Astro data plane. This allows you to run individual tasks on infrastructure that might not be supported on Astro yet, such as GPU nodes or on other third-party services.
```

A bit out of scope for this PR, but while we're here I added 1 more use case for the KPO in light of last week's bug fix re: allowing users to run tasks outside of Astro data plane
